benchmarks/metrics/counter.zig

```diff
@@ -0,0 +1,324 @@
+const std = @import("std");
+const sdk = @import("opentelemetry-sdk");
+const MeterProvider = sdk.MeterProvider;
+const benchmark = @import("benchmark");
+
+// Thread-local random number generator
+threadlocal var thread_rng: ?std.Random.DefaultPrng = null;
+
+fn getThreadRng() *std.Random.DefaultPrng {
+    if (thread_rng == null) {
+        thread_rng = std.Random.DefaultPrng.init(@as(u64, @intCast(std.time.timestamp())));
+    }
+    return &thread_rng.?;
+}
+
+// Benchmark configuration
+const bench_config = benchmark.Config{
+    .max_iterations = 100000,
+    .time_budget_ns = 2 * std.time.ns_per_s,
+    .track_allocations = false,
+};
+
+fn setupSDK(allocator: std.mem.Allocator) !*sdk.MeterProvider {
+    const mp = try MeterProvider.init(allocator);
+    errdefer mp.shutdown();
+    return mp;
+}
+
+// Generate random attributes similar to Rust benchmarks
+const ATTR_VALUES = [_][]const u8{
+    "value_0", "value_1", "value_2", "value_3", "value_4",
+    "value_5", "value_6", "value_7", "value_8", "value_9",
+};
+
+// Helper struct for pre-generated random indices
+const RandomIndicesPool = struct {
+    indices: [][4]u8,
+    counter: std.atomic.Value(u32),
+    allocator: std.mem.Allocator,
+
+    pub fn init(allocator: std.mem.Allocator, pool_size: usize) !RandomIndicesPool {
+        const indices = try allocator.alloc([4]u8, pool_size);
+        errdefer allocator.free(indices);
+
+        // Fill the pool with random indices
+        const rng = getThreadRng();
+        for (indices) |*idx_set| {
+            idx_set[0] = rng.random().intRangeAtMost(u8, 0, 9);
+            idx_set[1] = rng.random().intRangeAtMost(u8, 0, 9);
+            idx_set[2] = rng.random().intRangeAtMost(u8, 0, 9);
+            idx_set[3] = rng.random().intRangeAtMost(u8, 0, 9);
+        }
+
+        return RandomIndicesPool{
+            .indices = indices,
+            .counter = std.atomic.Value(u32).init(0),
+            .allocator = allocator,
+        };
+    }
+
+    pub fn deinit(self: *RandomIndicesPool) void {
+        self.allocator.free(self.indices);
+    }
+
+    pub fn getNext(self: *RandomIndicesPool) [4]u8 {
+        const idx = self.counter.fetchAdd(1, .monotonic) % self.indices.len;
+        return self.indices[idx];
+    }
+};
+
+test "Counter_Add_Sorted" {
```

Comment:
Looks like one test is gone: the one where `.add()` is called without any attributes.
That's the super-fast path. Am I correct we don't have that anymore?

Not a blocker to approve this Pr, given the effort you already put in ❤️